src/sql/workbench/browser/parts/views/customView.ts

```diff
@@ -700,6 +700,9 @@ class TreeDataSource implements IAsyncDataSource<ITreeItem, ITreeItem> {
 				// So in order to enable this we need to tell the tree to refresh this node so it will ask us for the data again
 				setTimeout(() => {
 					this.treeView.collapse(node);
+					if (e.message === 'User canceled') {
```

Comment:
A better solution I'd say would be to have the error thrown at https://github.com/microsoft/azuredatastudio/blob/ea0f9e6ce95b45def2b694efcd8ffe909bc43d20/src/sql/workbench/parts/objectExplorer/browser/objectExplorerViewTreeShim.ts#L93 be a custom error class instead and check for that.

It seems like having cancel be an error at all is kind of weird (I'd think just returning undefined would be fine), but that's not something I think we should be changing at this time so doing a quick check here is a fine enough solution for now I'd say.

Reply:
Definitely a better solution here, thanks @Charles-Gagnon.